server/src/controllers/course.ts

```diff
@@ -26,17 +28,57 @@ export enum Language {
   Swedish = 'SV'
 }
 
+interface JsonError extends Error {
+  expose?: boolean,
+  status?: number,
+  statusCode?: number,
+  body?: string,
+  type?: string
+}
+
+// Middleware function for handling errors in JSON parsing
```

Comment:
This seems to be the first PR implementing a custom middleware, should we put the middlewares to their own folder right from the beginning?

Reply:
Might be a good idea in future. Should we discuss this in the beginning of next sprint, while designing the common error handling more in detail?

Reply:
I think we should discuss this in the next sprint and decide on uniform conventions to follow.

Reply:
fair enough